libiocage/lib/Jails.py

```diff
@@ -46,9 +46,11 @@ def __init__(self,
                  logger=None,
                  zfs=None):
 
-        libiocage.lib.helpers.init_logger(self, logger)
-        libiocage.lib.helpers.init_zfs(self, zfs)
-        libiocage.lib.helpers.init_host(self, host)
+        self.logger = libiocage.lib.helpers.init_logger(logger)
+        self.host = libiocage.lib.helpers.init_host(host, logger=self.logger)
```

Comment:
Here's one example why we should pass self as first argument to the init_ helpers.